Mathlib/Algebra/Algebra/Basic.lean

```diff
@@ -330,7 +330,10 @@ theorem algebra_compatible_smul (r : R) (m : M) : r • m = (algebraMap R A) r 
 theorem algebraMap_smul (r : R) (m : M) : (algebraMap R A) r • m = r • m :=
   (algebra_compatible_smul A r m).symm
 
-theorem intCast_smul {k V : Type*} [CommRing k] [AddCommGroup V] [Module k V] (r : ℤ) (x : V) :
+lemma natCast_smul {k V : Type*} [Semiring k] [AddCommMonoid V] [Module k V] (n : ℕ) (x : V) :
+    (n : k) • x = n • x := algebraMap_smul k n x
```

Comment:
bors r-

We already have this lemma, its `nsmul_eq_smul_cast`. Can you deprecate the other one before creating this new one?